app/graphql/resolvers/top_volunteer_resolver.rb

```diff
@@ -0,0 +1,79 @@
+module TopVolunteerResolver
+  class << self
+    def all(_object, args, context)
+      user_entry = '"users"."id", "users"."first_name", "users"."last_name", "users"."email", "users"."group", "users"."photo"'
+
+      events_scope = User.joins(:events).select("#{user_entry}, \"events\".\"duration\"")
+      individual_events_scope = User.joins(:individual_events).select("#{user_entry}, \"individual_events\".\"duration\"")
+
+      office_id = case args[:office_id]
+                  when 'all'
+                    nil
+                  when 'current'
+                    context[:current_user].office_id
+                  else
+                    args[:office_id]
+                  end
+
+      events_scope, individual_events_scope = scope_with_time(events_scope, individual_events_scope, args[:after], args[:before])
+      events_scope, individual_events_scope = scope_with_office_id(events_scope, individual_events_scope, office_id)
+
+      all_events_scope = events_scope.union_all(individual_events_scope)
+      all_events_scope = all_events_scope.select("#{user_entry},  SUM(\"duration\") AS \"duration\"").group(user_entry)
+
+      all_events_scope = scope_with_count(all_events_scope, args[:count])
+      all_events_scope = scope_with_sort_by(all_events_scope, args[:sort_by])
+
+      all_events_scope
```

Comment:
If all the scopes are applied, this will look something like:

```
SELECT  "users"."id", "users"."first_name", "users"."last_name", "users"."email", "users"."group", "users"."photo",  SUM("duration") AS "duration" 
FROM (
  (
    SELECT "users"."id", "users"."first_name", "users"."last_name", "users"."email", "users"."group", "users"."photo", "events"."duration" 
    FROM "users" 
    INNER JOIN "signups" ON "signups"."user_id" = "users"."id" 
    INNER JOIN "events" ON "events"."id" = "signups"."event_id" 
    WHERE "users"."deleted_at" IS NULL 
    AND ("events"."starts_at" > '2018-12-31 13:00:00') 
    AND ("events"."starts_at" < '2019-10-09 01:38:01') 
    AND "users"."office_id" = $1
  ) 
  UNION ALL 
  (
    SELECT "users"."id", "users"."first_name", "users"."last_name", "users"."email", "users"."group", "users"."photo", "individual_events"."duration" 
    FROM "users" 
    INNER JOIN "individual_events" ON "individual_events"."user_id" = "users"."id" 
    AND "individual_events"."deleted_at" IS NULL 
    WHERE "users"."deleted_at" IS NULL 
    AND ("individual_events"."date" > '2018-12-31 13:00:00') 
    AND ("individual_events"."date" < '2019-10-09 01:38:01') 
    AND "users"."office_id" = $2
  ) 
) "users" 
GROUP BY "users"."id", "users"."first_name", "users"."last_name", "users"."email", "users"."group", "users"."photo" 
ORDER BY "duration" DESC 
LIMIT $3
```